ml/base/config.py

```diff
@@ -46,3 +37,36 @@ def __init__(self, global_config, graph, trainer, predictor, io, metric):
         self.predictor = predictor
         self.io = io
         self.metric = metric
+
+    @classmethod
+    def from_internal_file(cls, config_name):
+        pass
+
+    @classmethod
+    def from_file(cls, fn):
+        '''
+        Args:
+            fn: filename of a yaml config file.
+        '''
+        with open(fn, 'r') as file:
+            content = yaml.load(file)
+        return cls.from_dict(content)
+
+    @classmethod
+    def from_dict(cls, dic):
+        '''
+        Args:
+            dict: dictionary of a config.
+        '''
+        global_config = munch.munchify(dic)
+        component_config_lst = [global_config]
+
+        for component in ['graph', 'trainer', 'predictor', 'io', 'metric']:
+            config = munch.munchify(getattr(global_config, component))
+            component_config_lst.append(config)
+
+        return GlobalConfig(*component_config_lst)
+
+
+class CompConfig(MLConfig):
```

Comment:
Can we add a doc string to this unimplemented class?

Reply:
I just deleted it